vite.config.ts

```diff
@@ -0,0 +1,39 @@
+import path from 'path';
+
+import { defineConfig, loadEnv } from 'vite';
+import react from '@vitejs/plugin-react';
+
+export default ({ mode }) => {
+  // Load app-level env vars to node-level env vars.
+  process.env = { ...process.env, ...loadEnv(mode, process.cwd()) };
+  return defineConfig({
+    define: {
+      'process.env': process.env,
+    },
+    base: './',
+    plugins: [react()],
+    server: {
+      open: true, // automatically open the app in the browser
+      port: parseInt(process.env.PORT ?? '3000'),
+    },
+    preview: {
+      port: parseInt(process.env.PORT ?? '3000'),
```

Comment:
Shouldn't the port be 3001? It seems to be used in e.g. docker-compose.yml and in the readme

Reply:
Ah, it seems you fixed this in a later commit.